### PR TITLE
device-injector: remove unreachable code.

### PR DESCRIPTION
**Pull request by CormickKneey**

### CODE REVIEW

**Comment by kad** [comment]
for code changes, no questions, but for go.mod changes, I think it should be not part of this PR.

File: examples/go.mod [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-go 1.18
+go 1.21
+
+toolchain go1.22.0
 
 require (
 	github.com/containerd/cgroups v1.0.3
 	github.com/containerd/nri v0.1.0
 	github.com/opencontainers/runtime-spec v1.0.3-0.20220825212826-86290f6a00fb
-	github.com/sirupsen/logrus v1.8.1
```
Comment (klihub) [inline examples/go.mod:9]: nit: this is an unrelated dependency change.

File: examples/go.mod [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+	github.com/sirupsen/logrus v1.9.3
```
Comment (klihub) [inline examples/go.mod:11]: nit: unrelated deps change.

File: examples/go.mod [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
 )
 
 require (
 	github.com/coreos/go-systemd/v22 v22.3.2 // indirect
 	github.com/docker/go-units v0.4.0 // indirect
 	github.com/godbus/dbus/v5 v5.0.4 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
-	golang.org/x/sys v0.13.0 // indirect
+	golang.org/x/sys v0.21.0 // indirect
```
Comment (klihub) [inline examples/go.mod:19]: nit: unrelated deps change.

File: pkg/api/update.go [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 package api
 
-//nolint
+// nolint
```
Comment (klihub) [inline pkg/api/update.go:19]: Uhmm... this is a machine-readable comment-directive, not a human-readable comment. I think it should literally be `//nolint`.

**Review by klihub — COMMENTED** [review]
The core changes, IOW the removal of unreachable code is fine. But there are also a few seemingly unrelated dependency changes and an incorrect-looking update to a comment-directive tossed in there as well. I wonder if the deps changes are unintentional and the directive change, too....

**Comment by CormickKneey** [comment]
> The core changes, IOW the removal of unreachable code is fine. But there are also a few seemingly unrelated dependency changes and an incorrect-looking update to a comment-directive tossed in there as well. I wonder if the deps changes are unintentional and the directive change, too....

well, it's unintentional change and i revert it...

**Review by klihub — APPROVED** [review]
LGTM.

**Review by samuelkarp — APPROVED** [review]
